### PR TITLE
Fix Getting Dashcard with parameter that has empty name should not fail

### LINKED ISSUE #24500: Corrupted dashboard filter makes it impossible to view the question (unless removed)
**Describe the bug**
This is basically a duplicate of #15279, but that issue was created when we were still using the old syntax. Unfortunately, the E2E repro for this was more of a placeholder and it wasn't updated when the syntax changed so we didn't catch this regression.

**Logs**
```
POST /api/dashboard/*/dashcard/*/card/*/query results in

Output of dashboard->resolved-params does not match schema: 
	 [0;33m  {"af72ce9c" {:name (not ("Non-blank string" ""))}} [0m 
```

**To Reproduce**
1. Admin > Data Model > Sample Database > People > make sure to set **Source as "list of values"** and **Name as "search box"**
2. Simple question > Sample Dataset > People, and save as "Q1" and add to dashboard "D1"
3. On dashboard "D1", create two Text/Categories filters (both dropdown) - first connected (let's call it `List filter`) to Q1.Source, second (`Search filter`) connected to Q1.Name, save dashboard.
4. Question will not render and will display `There was a problem displaying this chart.`
5. `List filter` will not show a dropdown but "Enter some text" input field
6. `Search filter` will show "Search by name" but it doesn't seem it can find any value
7. Thanks to #16582 we can now remove the corrupted filter, so let's edit dashboard
8. Click on the gear icon next to the corrupted filter
image
9. You will notice it is possible to select question fields and to connect the filter to it (not sure if this should even be possible or should we just say "No fields available")
image
10. Remove filter from the sidebar (notice the question still doesn't display correctly) and save the dashboard
image
11. Either refresh or click on any of the working filters and set them, which will reset the corrupted question query
image

**Expected behavior**
Either the backend should make `name` required (and prevent adding a corrupted filter in the first place) or frontend should handle corrupted filter gracefully. @daltojohnso already made it possible to remove such filter in #16582. However, we should display the underlying question, especially since corrupted filter isn't even connected to the question.

**Screenshots**
image

**Information about your Metabase Installation:**
local dev, `master`, 3fc791200659436eadee74e2f2b12adc58f1dd9c, H2, Sample Database

**Severity**
I am not aware of the possibility to end up with a corrupted filter using the UI. It can only affect people trying to add filters via API and then forgetting to set the `name`.

**Additional context**
If filter cannot work without `name`, doesn't it make sense then to make that parameter required on the backend?

Why is it possible to send this payload (this adds corrupted filter to the dashboard)?
`PUT /api/dashboard/:id`
```json
{
  "name": "",
  "slug": "",
  "id": "af72ce9c",
  "type": "string/=",
  "sectionId": "string"
}
```

### PR DESCRIPTION
**Pull request by qnkhuat**
Fixes #24500 

Even though the bug is already fixed because FE will replace an empty name with `unnamed`.

But BE will still fail if somehow parameters have an empty name (this could be done if updated via API).

This PR changes so that executing a dashcard with parameter that has empty name won't fail. `name` is never a required field, only `type` and `id` is. Check the schema for Parameter [here](https://github.com/metabase/metabase/blob/892e85a9a62afa9c4e4327d78cb440f50a80a6ce/src/metabase/util/schema.clj#L579)

### CODE REVIEW

File: src/metabase/models/dashboard.clj [hunk 1]
```diff
@@ -381,10 +381,10 @@
     dashboard))
 
 (def ^:private ParamWithMapping
-  {:name     su/NonBlankStringPlumatic
-   :id       su/NonBlankStringPlumatic
-   :mappings (s/maybe #{dashboard-card/ParamMapping})
-   s/Keyword s/Any})
+  {:id                    su/NonBlankStringPlumatic
```
Comment (qnkhuat) [inline src/metabase/models/dashboard.clj:384]: the bug from 24500 is from when calling `dashboard->resolved-params*`, it expects output that has a nonblankstring name.

As described in the PR's description. Name is not required, so we will mark them as optional here.

**Comment by qnkhuat** [comment]
@nemanjaglumac could you please help me with the failing Cypress test?

FYI:
- Parameter with empty name is a valid parameter, it's not corrupted
- If we create a parameter with empty name using UI, FE will automatically fill the name with `unnamed`

File: test/metabase/models/dashboard_test.clj [hunk 2]
```diff
@@ -408,3 +409,23 @@
         (is (= "8cbf93b7"
                (serdes.hash/raw-hash ["my dashboard" (serdes.hash/identity-hash c1) now])
                (serdes.hash/identity-hash dash)))))))
+
+(deftest parameter-with-null-name-is-allowed-test
+  (testing "parametesr with null or empty name should works (#24500)"
```
Comment (tsmacdonald) [inline test/metabase/models/dashboard_test.clj:414]:
```suggestion
  (testing "parameters with null or empty names should work (#24500)"
```

**Review by tsmacdonald — APPROVED** [review]
Does what it says on the tin :boat:

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
Making the schema (even more) loose leads to a different can of worms.

> Even though the bug is already fixed because FE will replace an empty name with unnamed.

See https://github.com/metabase/metabase/issues/27653 for more details.
I'll repeat the important bit here: FE uses `name` to derive a `slug`, and without a `slug` it's impossible to distinguish between multiple unnamed filters in the query parameters.

As @qnkhuat and I saw during the video chat, leaving both `name` and the `slug` empty (this is only possible using the API and bypassing FE) can have really ugly consequences.

1. query parameters will be set without a slug, and only with a value like so `?=foo`
2. on a dashboard refresh, this filter value will be dropped

Blocking the merge until we discuss how to approach this problem and whether it's worth protecting against unnamed filters in the first place.

**Comment by qnkhuat** [comment]
as @nemanjaglumac pointed out, the approach of this PR does not solve the root cause.
We should make parameter's  name required and unique on both FE and BE and close it https://github.com/metabase/metabase/issues/27653 altogether.

**Pull request closed without merging**